net-misc/kea/kea-9999.ebuild

```diff
@@ -0,0 +1,260 @@
+		-i src/bin/shell/kea-shell.in || die
+
+	# Don't allow meson to install shell, we shall do that if required
+	sed -e 's:install\: true:install\: false:' \
+		-i src/bin/shell/meson.build || die
+
+	# do not create /run
+	sed -e '/^install_emptydir(RUNSTATEDIR)$/d' \
+		-i meson.build || die
+}
+
+src_configure() {
+	# https://bugs.gentoo.org/861617
+	# https://gitlab.isc.org/isc-projects/kea/-/issues/3946
+	#
+	# Kea Devs say no to LTO
+	filter-lto
+
+	local emesonargs=(
+		--localstatedir="${EPREFIX}/var"
+		-Drunstatedir="${EPREFIX}/run"
+		-Dkrb5=disabled
+		-Dnetconf=disabled
+		-Dcrypto=$(usex openssl openssl botan)
+		$(meson_feature mysql)
+		$(meson_feature postgres postgresql)
+		$(meson_feature test tests)
+	)
+	if use debug; then
+		emesonargs+=(
+			--debug
+		)
+	fi
+	meson_src_configure
+}
+
+src_test() {
+	# Get list of all test suites into an associative array
+	# the meson test --list returns either "kea / test_suite", "kea:shell-tests / test_suite" or
+	# "kea:python-tests / test_suite"
+	# Discard the shell tests as we can't run shell tests in sandbox
+
+	pushd "${BUILD_DIR}" || die
+	local -A TEST_SUITES
+	while IFS=" / " read -r subsystem test_suite ; do
+		if [[ ${subsystem} != "kea:shell-tests" ]]; then
+			TEST_SUITES["$test_suite"]=1
+		fi
+	done < <(meson test --list || die)
+	popd
+
+	# Some other tests will fail for interface access restrictions, we have to remove the test suites those tests
+	# belong to
+	local SKIP_TESTS=(
+		dhcp-radius-tests
+		kea-log-buffer_logger_test.sh
+		kea-log-console_test.sh
+		dhcp-lease-query-tests
+		kea-dhcp6-tests
+	)
+
+	if [[ $(tc-get-ptr-size) -eq 4 ]]; then
+		# see https://bugs.gentoo.org/958171 for reason for skipping these tests
+		SKIP_TESTS+=(
+			kea-util-tests
+			kea-dhcp-tests
+			kea-dhcp4-tests
+			kea-dhcpsrv-tests
+			dhcp-ha-lib-tests
+			kea-d2-tests
+		)
+	fi
+
+	for SKIP in ${SKIP_TESTS[@]}; do
+		unset TEST_SUITES["${SKIP}"]
+	done
+
+	meson_src_test ${!TEST_SUITES[@]}
+}
+
+install_shell() {
+	python_domodule ${ORIG_BUILD_DIR}/src/bin/shell/*.py
+	python_doscript ${ORIG_BUILD_DIR}/src/bin/shell/kea-shell
+
+	# fix path to import kea modules
+	sed -e "/^sys.path.append/s|(.*)|('$(python_get_sitedir)/${PN}')|"	\
+		-i "${ED}"/usr/lib/python-exec/${EPYTHON}/kea-shell || die
+}
+
+src_install() {
+	meson_install
+
+	# No easy way to control how meson_install sets permissions in meson < 1.9
+	# So make sure permissions are same as in previous versions of kea
+	# To avoid any differences between an update vers first time install
+	fperms -R 0755 /usr/sbin
+	fperms -R 0755 /usr/bin
+	fperms -R 0755 /usr/$(get_libdir)
+
+	if use shell; then
+		python_moduleinto ${PN}
+		ORIG_BUILD_DIR=${BUILD_DIR} python_foreach_impl install_shell
+	fi
+
+	dodoc -r doc/examples
+
+	diropts -m 0750 -o root -g dhcp
+	dodir /etc/kea
+	insopts -m 0640 -o root -g dhcp
+	insinto /etc/kea
+	newins doc/examples/agent/comments.json kea-ctrl-agent.conf.sample
```

Comment:
new build system installs now config files with wrong permissions root:root
I don't know what's the best: keep only the sample or leave only .conf